From aaa145f871aa5626378bcd767554f49f2f15705e Mon Sep 17 00:00:00 2001
From: Fabio Berton <fabio.berton@ossystems.com.br>
Date: Tue, 25 Jul 2017 13:34:45 -0300
Subject: [PATCH] BaseTools/Source/C/Makefiles/header.makefile: Add native
 header and libs
Organization: O.S. Systems Software LTDA.

We need to add STAGING_INCDIR_NATIVE and STAGING_LIBDIR_NATIVE
to build edk, whitout these gcc can't find ossp-uuid header and lib.

Upstream-Status: Pending

Signed-off-by: Fabio Berton <fabio.berton@ossystems.com.br>
---
 BaseTools/Source/C/Makefiles/header.makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/BaseTools/Source/C/Makefiles/header.makefile b/BaseTools/Source/C/Makefiles/header.makefile
index 063982b82..9b8f72bbf 100644
--- a/BaseTools/Source/C/Makefiles/header.makefile
+++ b/BaseTools/Source/C/Makefiles/header.makefile
@@ -43,7 +43,7 @@ ifeq ($(ARCH), AARCH64)
 ARCH_INCLUDE = -I $(MAKEROOT)/Include/AArch64/
 endif
 
-INCLUDE = $(TOOL_INCLUDE) -I $(MAKEROOT) -I $(MAKEROOT)/Include/Common -I $(MAKEROOT)/Include/ -I $(MAKEROOT)/Include/IndustryStandard -I $(MAKEROOT)/Common/ -I .. -I . $(ARCH_INCLUDE) 
+INCLUDE = $(TOOL_INCLUDE) -I $(MAKEROOT) -I $(MAKEROOT)/Include/Common -I $(MAKEROOT)/Include/ -I $(MAKEROOT)/Include/IndustryStandard -I $(MAKEROOT)/Common/ -I .. -I . $(ARCH_INCLUDE) -I${STAGING_INCDIR_NATIVE}
 BUILD_CPPFLAGS = $(INCLUDE) -O2
 ifeq ($(DARWIN),Darwin)
 # assume clang or clang compatible flags on OS X
@@ -51,7 +51,7 @@ BUILD_CFLAGS = -MD -fshort-wchar -fno-strict-aliasing -Wall -Werror -Wno-depreca
 else
 BUILD_CFLAGS = -MD -fshort-wchar -fno-strict-aliasing -Wall -Werror -Wno-deprecated-declarations -Wno-unused-result -nostdlib -c -g
 endif
-BUILD_LFLAGS =
+BUILD_LFLAGS = ${BUILD_LDFLAGS} -L${STAGING_LIBDIR_NATIVE}
 BUILD_CXXFLAGS = -Wno-unused-result
 
 ifeq ($(ARCH), IA32)
-- 
2.13.3

